ui/app/ducks/swaps/swaps.js

```diff
@@ -530,3 +613,47 @@ export const signAndSendTransactions = (history, metaMetricsEvent) => {
     await forceUpdateMetamaskState(dispatch)
   }
 }
+
+export function fetchMetaSwapsGasPriceEstimates () {
+  return async (dispatch, getState) => {
+    const state = getState()
+    const priceEstimatesLastRetrieved = getSwapsPriceEstimatesLastRetrieved(state)
+    const timeLastRetrieved = priceEstimatesLastRetrieved || loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES_LAST_RETRIEVED') || 0
+
+    dispatch(swapGasPriceEstimatesFetchStarted())
+
+    let priceEstimates
+    try {
+      if (Date.now() - timeLastRetrieved > 30000) {
+        priceEstimates = await fetchSwapsGasPrices()
+      } else {
+        const cachedPriceEstimates = loadLocalStorageData('METASWAP_GAS_PRICE_ESTIMATES')
+        priceEstimates = cachedPriceEstimates || await fetchSwapsGasPrices()
+      }
+    } catch (e) {
+      dispatch(swapGasPriceEstimatesFetchFailed())
+
+      try {
+        const gasPrice = await global.ethQuery.gasPrice()
+        const gasPriceInDecGWEI = hexWEIToDecGWEI(gasPrice.toString(10))
+
+        dispatch(retrievedFallbackSwapsGasPrice(gasPriceInDecGWEI))
+
+        return null
+      } catch (e2) {
+        return null
+      }
```

Comment:
Two suggestions:

* I don't know what failures to expect here offhand, and if we've gotten here something is quite wrong anyway, so printing the error to the console at least should give some indication if we need to debug this, or if a user sends this console log in.
* `finally` is nice when the last steps are the same for both cases.
```suggestion
      } catch (networkGasPriceError) {
        console.error(`Failed to retrieve fallback gas price: `, networkGasPriceError)
      } finally {
        return null
      }
```

Reply:
had to remove the finally because
![Screenshot from 2020-10-23 14-27-46](https://user-images.githubusercontent.com/7499938/97032187-fb66bc80-153b-11eb-8022-b957aa7c0703.png)